From 5af80d41384c05ad92464682f9771fd75590fea2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tommi=20My=C3=B6h=C3=A4nen?= <ext-tommi.1.myohanen@nokia.com>
Date: Thu, 13 Aug 2009 10:33:07 +0300
Subject: [PATCH 03/23] camerabin: change video viewfinder queue to be leaky
 upstream (Maemo specific)

Leaking upstream buffers instead of downstream ones seem to give a small
performance boost for video recording.
---
 gst/camerabin/camerabinvideo.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gst/camerabin/camerabinvideo.c b/gst/camerabin/camerabinvideo.c
index a98807b..08a842e 100644
--- a/gst/camerabin/camerabinvideo.c
+++ b/gst/camerabin/camerabinvideo.c
@@ -624,7 +624,7 @@ gst_camerabin_video_create_elements (GstCameraBinVideo * vid)
   }
   /* Set queue leaky, we don't want to block video encoder feed, but
      prefer leaking view finder buffers instead. */
-  g_object_set (G_OBJECT (queue), "leaky", 2, "max-size-buffers", 1, "silent",
+  g_object_set (G_OBJECT (queue), "leaky", 1, "max-size-buffers", 1, "silent",
       TRUE, NULL);
 
   /* Set up src ghost pad for video bin */
-- 
1.7.10.4

